common/src/main/java/com/commercetools/sunrise/payment/domain/PaymentServiceProvider.java

```diff
@@ -0,0 +1,48 @@
+package com.commercetools.sunrise.payment.domain;
+
+import io.sphere.sdk.payments.Payment;
+import io.sphere.sdk.payments.PaymentMethodInfo;
+import io.sphere.sdk.payments.PaymentStatus;
+
+import java.util.List;
+import java.util.Map;
+import java.util.function.BiFunction;
+import java.util.function.Function;
+
+/**
+ * A payment service provider provides functions that can be used to handle payment related tasks in
+ * a PSP specific way.
+ * @author mgatz
+ */
+public interface PaymentServiceProvider {
+
+    /**
+     * @return a unique identifier of the payment service provider
+     */
+    String getId();
+
+    /**
+     * @return a list of payment methods this PSP supports
+     */
+    List<PaymentMethodInfo> getAvailablePaymentMethods(); // TODO decide parameters
```

Comment:
Possible parameters could be custom-object ```container``` and ```key```. I think if the shop has own configuration whatever format it has then it can load it by itself without using this service, wdyt?